### ISSUE TITLE
How do I configure what port will be opened in the browser when clicking on "Open in Browser" in the dashboard

### ISSUE THREAD

**Issue by chrisbra** [docker/for-win#9537]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [ ] I have uploaded Diagnostics - not relevant 
  - Diagnostics ID:

### Expected behavior

I have a docker-compose file, that exposes several ports, among there are 2 different http ports for accessing the docker container via a browser interface. However unfortunately, I cannot find out what rules are used by docker-desktop dashboard to find out which port to open in a browser. I checked the documentation at https://docs.docker.com/desktop/dashboard/ but it doesn't mention it 

I already tried a different ordering of the ports section from my docker-compose yaml file. But this does not seem to matter.

Any suggestions?

Thanks,
Christian

**Comment by janhh** [comment]
I struggled with the same thing.

It appears to me that the docker-desktop just goes with the lowest port number on the container.

After I changed the port number for the conflicting service on the container to a value higher than the http service, I can now use the "open in browser" link from docker-desktop dashboard. I assume that the port numbers are sorted numerically and not by string value, but I could not be certain without further testing.

**Comment by lifan0127** [comment]
Ability to customize the port would make this feature truly useful.

**Comment by robertsearle** [comment]
Actually I would like the ability to specify a default URL and a list of urls
E.G.
defaultUrl: https://localhost:9443/ibmmq/console

```
urls:
  - https://localhost:9443/ibmmq/console (nice if it auto includes above)
  - https://localhost:9443/ibmmq/console2
  - https://localhost:1234/something interesting
```

**Comment by robertsearle** [comment]
Actually I would like the ability to specify a default URL and a list of urls
E.G.
defaultUrl: https://localhost:9443/ibmmq/console

```
urls:
  - https://localhost:9443/ibmmq/console (nice if it auto includes above)
  - https://localhost:9443/ibmmq/console2
  - https://localhost:1234/something interesting
```

/remove-lifecycle stale

**Comment by robertsearle** [comment]
Actually I would like the ability to specify a default URL and a list of urls
E.G.
defaultUrl: https://localhost:9443/ibmmq/console

```
urls:
  - https://localhost:9443/ibmmq/console (nice if it auto includes above)
  - https://localhost:9443/ibmmq/console2
  - https://localhost:1234/something interesting
```

`/remove-lifecycle stale`

**Comment by robertsearle** [comment]
Actually I would like the ability to specify a default URL and a list of urls
E.G.
defaultUrl: https://localhost:9443/ibmmq/console

```
urls:
  - https://localhost:9443/ibmmq/console (nice if it auto includes above)
  - https://localhost:9443/ibmmq/console2
  - https://localhost:1234/something interesting
```

/remove-lifecycle stale

**Comment by crocokyle** [comment]
Agreed, even running it as the first port in the Dockerfile would be better than the lowest port number. Sometimes port reassignment isn't an option

**Comment by chrisbra** [comment]
This is not solved and still open.

**Comment by mwt** [comment]
I agree that it should be the first port binding specified rather than the lowest port number

**Comment by ptn77** [comment]
Hi 
I'm having the same issue here. where can I update the url in the docker desktop in order to open in browser? I ran the docker container through the visual studio IDE and I see the port numbers when I run docker ps, but when I try to access it from the browser using the port number displayed by docker, it does not work.

**Comment by its-mash** [comment]
any update?

**Comment by its-mash** [comment]
@chrisbra  can you mark  "I have uploaded Diagnostics - not relevant" done, seems otherwise  it won't reach them. and bot will keep closing

**Comment by chrisbra** [comment]
@its-mash no, there were no bot messages here for a long time. And I don't think it helps if I pretend that I have send some diagnostics which do not even exists. 

Also seriously, there hasn't been any messages from developers here. So I personally do not hope that this will be fixed (or rather properly documented) any time soon.

**Comment by its-mash** [comment]
I guess they aren't notified yet as the issue is in draft stage unless you mark the checklist done. As you don't need diagnostics here, u can mark it done.

**Comment by mccreah** [comment]
This is still an issue. VS assigns one port number for https and the other for http, with https one being lower than http port. 

When you click Open in browser in Docker desktop it will open the lower port number in browser with http protocol obviously resulting in error since you cannot browse to https port using http protocol (we've established the lower port being https in most cases when using Visual Studio web apps). 

The other way around works bc the app is configured so that what you browse to the http port it will redirect to the https.

If Docker desktop opened the app with http protocol against http port it would be fine. It would also be fine if it opened the app with https protocol against https port. 

But there is no way it will work opening an app with http protocol against https port.

The Inspect tab clearly shows which port maps to which protocol. So for 80 you take the correct port. **You DO NOT take the 443 port and open it with http://**

![docker-ports](https://user-images.githubusercontent.com/110022491/186675844-419c1fcf-e3ba-4d9a-9b63-822b0b52a720.png)

Docker team should fix this asap.

**Comment by henkdegroot** [comment]
Same here, would love to be able to specify which port to be opened using https. Even when I set the port 443:443, the link in docker desktop still show http://localhost:443/....something must be wrong here.

**Comment by Getinge-Dumitru** [comment]
Same issue here and after 2 years 0 progress?
